### PR TITLE
solvers

### LINKED ISSUE #9: Difference between static / regular arrays
```julia
julia> A = @SMatrix [2..4 -2..1; -1..2 2..4]
julia> b = @SVector [-2..2, -2..2];

julia> Am = Matrix(A)
julia> bm = Vector(b);

julia> solve(A, b, Jacobi())
solve(A, b, Jacobi()) = Interval{Float64}[[-14.0001, 14.0001], [-14.0001, 14.0001]]

julia> solve(Am, bm, Jacobi())
solve(Am, bm, Jacobi()) = Interval{Float64}[[-45.7501, 45.7501], [-45.0795, 45.0795]]
```

### PR DESCRIPTION
**Pull request by lucaferranti**
* Make sure solvers behave the same way for both static matrices and normal matrices (fixes #9 )

* added tests for linear solvers

I need to investigate the failure on nightly

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1,6 +1,29 @@
-using IntervalLinearAlgebra
+using IntervalLinearAlgebra, StaticArrays, IntervalArithmetic
 using Test
 
+int_approx(a, b, tol) =  abs(a.lo - b.lo) < tol  && abs(a.hi - b.hi) < tol
```
Comment (dpsanders) [inline test/runtests.jl:4]: Int here sounds like integer. Maybe interval_approx?

Comment (lucaferranti) [reply test/runtests.jl:4]: good point, although I guess the best solution would be to open a PR to IntervalArithmetic.jl and overload `isapprox` there?

File: src/utils.jl [hunk 2]
```diff
@@ -13,7 +13,7 @@ that is the midpoint of A.
 """
 function precondition(A, b)
     Ac = mid.(A)
-    return Ac\A, Ac\b
+    return inv(Ac)*A, inv(Ac)*b
```
Comment (mforets) [inline src/utils.jl:16]: This computes the inverse two times.

Comment (lucaferranti) [reply src/utils.jl]: fixed!

File: test/runtests.jl (hunk 1 above)
Comment (lucaferranti) [reply test/runtests.jl]: for the time being, I've overloaded `isapprox` here. This should be moved to `IntervalArithmetic.jl` at some point I guess

**Comment by lucaferranti** [comment]
@dpsanders @mforets this should now be ready to be merged. In a nutshell, this PR has reorganized the solvers that were already implemented, particularly

1) added pretty printing of solvers and more details in docstrings.
2) avoid using backslash internally
3) Fixed bug in formula for `enclose`
4) Overloaded `Base.isapprox` to use tolerances in iterative algorithms.

File: src/utils.jl [hunk 3]
```diff
@@ -12,36 +14,30 @@ Preconditions the interval system Ax = b by multipling by the (approximate) inve
...
-end
-
-function enclose(A::MMatrix{N, N, T, M}, b::MVector{N, T}) where {N, T, M}
-    A1 = Diagonal(ones(N)) - A
-    e = interval_norm(b)/(1 - interval_norm(A1))
+function enclose(A::StaticMatrix{N, N, T}, b::StaticVector{N, T}) where {N, T}
+    C = inv(mid.(A))
+    A1 = Diagonal(ones(N)) - C*A
+    e = interval_norm(C*b)/(1 - interval_norm(A1))
     x0 = MVector{N, T}(fill(-e..e, N))
```
[the remark this answers is not in the document]
Comment (dpsanders) [reply src/utils.jl:32]: I think using the `ntuple` function

[the remark this answers is not in the document]
Comment (lucaferranti) [reply src/utils.jl]: it indeed reduced the allocations a little

**Comment by lucaferranti** [comment]
@dpsanders @mforets can this be merged?

**Review by mforets — APPROVED** [review]

**Pull request merged**